### PR TITLE
Use v8StrIntern more to reduce number of string copies in v8 heap

### PR DESCRIPTION
**Pull request by jasnell**
Generally non-functional change.

### CODE REVIEW

File: src/workerd/api/urlpattern.c++ [hunk 1]
```diff
@@ -390,7 +390,7 @@ bool protocolComponentMatchesSpecialScheme(jsg::Lock& js, URLPatternComponent& c
   auto context = js.v8Context();
 
   const auto checkIt = [&handle, &js, &context](const char* name) {
-    return !jsg::check(handle->Exec(context, jsg::v8Str(js.v8Isolate, name)))->IsNullOrUndefined();
+    return !jsg::check(handle->Exec(context, jsg::v8StrIntern(js.v8Isolate, name)))->IsNullOrUndefined();
```
Comment (dom96) [inline src/workerd/api/urlpattern.c++:393]: nit: wrap line

File: src/workerd/jsg/modules.c++ [hunk 2]
```diff
@@ -371,7 +371,7 @@ v8::Local<v8::Module> compileEsmModule(
     // TODO(later): Use of newExternalOneByteString here limits our built-in source
     // modules (for which this path is used) to only the latin1 character set. We
     // may need to revisit that to import built-ins as UTF-16 (two-byte).
-    contentStr = jsg::check(jsg::newExternalOneByteString(js, content));
+    contentStr = jsg::newExternalOneByteString(js, content);
```
Comment (dom96) [inline src/workerd/jsg/modules.c++:374]: why is the `jsg::check` no longer necessary?

**Review by dom96 — APPROVED** [review]

File: src/workerd/jsg/modules.c++ (hunk 2 above)
Comment (jasnell) [reply src/workerd/jsg/modules.c++:374]: The PR moves the jsg::check into the implementaiton of `newExternalOneByteString` to simplify

File: src/workerd/api/urlpattern.c++ (hunk 1 above)
Comment (jasnell) [reply src/workerd/api/urlpattern.c++:393]:
```suggestion
    return !jsg::check(handle->Exec(context, jsg::v8StrIntern(js.v8Isolate, name)))
        ->IsNullOrUndefined();
```

**Review by ohodson — APPROVED** [review]

**Comment by kentonv** [comment]
Is there any documentation or guidance for when strings should be internalized? I don't feel like I have any intuition for this.

**Comment by ohodson** [comment]
Not for any runtime that I'm aware of. The biggest win usually comes from places where code will compare two strings that are likely to interned. Interning has some cost, but if comparisons are frequent it's a win.

In this PR, most of these are saving some GC work. Whether this is actually a perf win would need some measurement (e.g. young generation is usually cheap to allocate and collect).

There is typically an upper bound on the size of the data structure holding interned strings so it usually requires some finessing.

**Pull request merged**